pocs/observatory.py

```diff
@@ -537,6 +555,235 @@ def close_dome(self):
+            camera_list (list, optional): List of cameras to use for flat-fielding
+            target_adu_percentage (float, optional): Exposure time will be adjust so
+                that counts are close to: target * (`min_counts` + `max_counts`). Default
+                to 0.5
+            max_num_exposures (int, optional): Maximum number of flats to take
+            take_darks (bool, optional): If dark fields matching flat fields should be
+                taken, default False.
+            *args (TYPE): Description
+            **kwargs (TYPE): Description
+        """
+        target_adu = target_adu_percentage * (min_counts + max_counts)
+
+        image_dir = self.config['directories']['images']
+
+        flat_obs = self._create_flat_field_observation(alt=alt, az=az)
+
+        if camera_list is None:
+            camera_list = list(self.cameras.keys())
+
+        # Start with 1 second exposures for each camera
+        exp_times = {cam_name: [1. * u.second] for cam_name in camera_list}
+
+        # Loop until conditions are met for flat-fielding
+        while True:
+            self.logger.debug("Slewing to flat-field coords: {}".format(flat_obs.field))
+            self.mount.set_target_coordinates(flat_obs.field)
+            self.mount.slew_to_target()
+            self.status()  # Seems to help with reading coords
+
+            fits_headers = self.get_standard_headers(observation=flat_obs)
+
+            start_time = current_time()
+            fits_headers['start_time'] = utils.flatten_time(start_time)
+
+            camera_events = dict()
+
+            # Take the observations
+            for cam_name in camera_list:
+
+                camera = self.cameras[cam_name]
+                exp_time = exp_times[cam_name][-1].value
+
+                filename = "{}/flats/{}/{}/{}.{}".format(
+                    image_dir,
+                    camera.uid,
+                    flat_obs.seq_time,
+                    'flat_{:02d}'.format(flat_obs.current_exp),
+                    camera.file_extension)
+
+                # Take picture and get event
+                camera_event = camera.take_observation(
+                    flat_obs,
+                    fits_headers,
+                    filename=filename,
+                    exp_time=exp_time
+                )
+
+                camera_events[cam_name] = {
+                    'event': camera_event,
+                    'filename': filename,
+                }
+
+            # Block until done exposing on all cameras
+            while not all([info['event'].is_set() for info in camera_events.values()]):
+                self.logger.debug('Waiting for flat-field image')
+                time.sleep(1)
+
+            # Check the counts for each image
+            is_saturated = False
+            for cam_name, info in camera_events.items():
+                img_file = info['filename'].replace('.cr2', '.fits')
```

Comment:
That appears to be a canon specific path. I thought this came from Huntsman?

Reply:
I adapted it first and tested it in PAN008 before submitting PR